src/core/server/elasticsearch/elasticsearch_service.ts

```diff
@@ -129,3 +128,28 @@ export class ElasticsearchService implements CoreService<InternalElasticsearchSe
     );
   }
 }
+
+const isMergable = (obj: any) => typeof obj === 'object' && obj !== null && !Array.isArray(obj);
+
+/** Deeply merges two objects, omitting undefined values and not deeply merging Arrays. */
+const merge = <T extends Record<string, any>, U extends Record<string, any>>(
```

Comment:
let's move the function in a separate file. that would simplify testing **a lot**

Reply:
Do you think we should move this to `src/core/utils` or just keep in here for now?

Reply:
here is fine. until we have one use-case more